File: 2022-2023/sourcecode/lab/lab5/MyFinanceAppJava/app/src/main/java/pl/edu/uwr/pum/myfinanceappjava/adapters/FinanceAdapter.java
package pl.edu.uwr.pum.myfinanceappjava.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.ArrayList;

import pl.edu.uwr.pum.myfinanceappjava.fragments.AccountsFragment;
import pl.edu.uwr.pum.myfinanceappjava.fragments.BillsFragment;
import pl.edu.uwr.pum.myfinanceappjava.fragments.OverviewFragment;

public class FinanceAdapter extends FragmentStateAdapter {

    private final Fragment[] fragments = {
            new OverviewFragment(),
            new AccountsFragment(),
            new BillsFragment()
    };

    public FinanceAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragments[position];
    }

    @Override
    public int getItemCount() {
        return fragments.length;
    }
}

File: 2022-2023/sourcecode/lab/lab5/MyFinanceAppKotlin/app/src/main/java/pl/edu/uwr/pum/myfinanceappkotlin/adapters/FinanceAdapter.kt
package pl.edu.uwr.pum.myfinanceappkotlin.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import pl.edu.uwr.pum.myfinanceappkotlin.fragments.AccountsFragment
import pl.edu.uwr.pum.myfinanceappkotlin.fragments.BillsFragment
import pl.edu.uwr.pum.myfinanceappkotlin.fragments.OverviewFragment

class FinanceAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
    private val fragments = listOf(OverviewFragment(), AccountsFragment(), BillsFragment())

    override fun getItemCount(): Int = fragments.size

    override fun createFragment(position: Int): Fragment = fragments[position]
}